### DISCUSSION TITLE
Tauri GlobalShortcut

### DISCUSSION THREAD

**Discussion by dev-mash** [tauri-apps/tauri#9550] [General]
Hi,
I added global shortcuts to my app recently. Now I found the following unexpected behavior:
- Global Shortcuts will be emitted even if the app is not the primary window
- The global shortcut will break the primary windows shortcuts

Example:
I open my app which registers ctrl+n
I open my webbrowser
I press ctrl+N
This should create a new window of my browser
The browser will not react to the shortcut anymore
Instead, the ctrl+n event is emitted in my app

Is this the expected behavior? If yes, how could I make sure to get the events only if my app is the primary window?

Best reards,
Manuel

**Comment by FabianLars** [comment]
This is expected, it is a **global** shortcut/hotkey after all. The currently only viable alternative is to use javascript key events to listen to the key combination.

**Reply by dev-mash** [reply]
Thanks for your answer, Fabian. I will try check the suggested solution with javascript key events.

PS: I think I never used a program that would be able to handle such a shortcuts and even overrule the shortcuts of the primary window.

**Reply by FabianLars** [reply]
A common example is something like push-to-talk in discord/skype/any voice chat program, or a screenshot tool (including OS built-in ones i guess). We don't do anything special really, we use the existing OS hotkey apis as they are.

**Reply by dev-mash** [reply]
Got it. I should probably not use global shortcuts for my use case. I switched now to javascript key events.

**Reply by dev-mash** [reply]
One follow up question: If I want to open my app with a global shortcut, how would I do that? The app would need to listen to the shortcut even if the window is closed. Is this possible?

**Reply by FabianLars** [reply]
well, that sounds like a use case for an actual global shortcut though of course it requires to not use a really common key combination but that also applies if you'd have a non blocking key listener.

**Reply by dev-mash** [reply]
Thats true, the key combination should not be too common.
Is there any example how this can be implemented? I think my app would needs to run to register the shortcuts. But currently my app opens a window and the app stops when the window is closed. I would need the app to run in the background without a window.

**Reply by FabianLars** [reply]
This is the code that will keep the event loop running: https://github.com/tauri-apps/tauri/blob/b02fc90f450ff9e9d8a35ee55dc1beced4957869/examples/api/src-tauri/src/main.rs#L242-L244

**Reply by dev-mash** [reply]
Thank you!
